qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -89,6 +92,10 @@ def __init__(self, params, num_qubits=None):
             if num_qubits == 0 or not num_qubits.is_integer():
                 raise QiskitError("Desired statevector length not a positive power of 2.")
 
+                # Check if normalize=True then normalizes input array
+            if normalize is True:
+                params = self._normalize(params)
+
             # Check if probabilities (amplitudes squared) sum to 1
             if not math.isclose(sum(np.absolute(params) ** 2), 1.0, abs_tol=_EPS):
                 raise QiskitError("Sum of amplitudes-squared does not equal one.")
```

Comment:
I know you're copying the style of the surrounding code which is good, but the comment is a bit unnecessary (several of the existing ones in this file are) - it's 100% clear from the well-named functions and variables that that's what's going on here, so the comment is just noise.

We also could skip the second test if `normalize=True`, because we'll already know for certain that the input is normalised to within floating-point tolerance.